ArjunaJTA/jms/src/main/java/org/jboss/narayana/jta/jms/JmsXAResourceRecoveryHelper.java

```diff
@@ -177,4 +268,11 @@ private void disconnect() throws XAException {
         }
     }
 
+    private void assertIfRecoveryScanWasStarted() throws XAException {
+        if (delegate == null) {
+            LOGGER.warn("XAResource.recover has to be called first with XAResource.TMSTARTRSCAN argument");
+            throw new XAException(XAException.XAER_PROTO);
```

Comment:
Whichever error code it is, I think we're too picky. Initially I had AssertionError. But I decided to remove it, because XARecoveryModule only catches XAException. So AssertionError used to crash the recovery module.
This scenario happened for me a few times on Spring during the shutdown when JMS connection is closed before RecoveryManager is stopped.